### PR TITLE
Added project dependencies

### PR DESCRIPTION
**Pull request by Hossam-Elgmmal**
## Description

This **PR** adds dependencies needed for the project and updates kotlin version to **1.9.23** to fix build issues with kapt.
It uses kapt with hilt because dagger’s ksp support is currently in alpha.

This ensures the project uses the latest tools and libraries, improving stability and maintainability.

---

### added dependencies for : 

* Hilt & kapt
* Retrofit & Okhttp & kotlinx Json
* Coil
* Proto datastore
* Room & ksp
* mockk & courotines testing
* compose navigation & hilt view model
* splash screen
* work manager

---

## Screenshots

> Not applicable.

---

## Checklist

Please ensure the following tasks are completed:

- [ ] Unit tests are included for the new functionality (if applicable).
- [x] Code has been properly documented.
- [x] Changes have been tested and verified.

---

## Additional Comments

> Add tests as you go.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -66,4 +73,94 @@ dependencies {
...
+    implementation(libs.kotlinx.serialization.json)
+
+    // Coil - asynchronous image loading
+    implementation(libs.coil.compose)
+
+    // datastore - make type safe local storage
+    implementation(libs.androidx.datastore)
+    // protobuf - serialize Kotlin objects <-> protobuf schema
+    implementation(libs.protobuf.kotlin.lite)
+
```
Comment (qamarelsafadi) [inline app/build.gradle.kts:100]: Can you explain what is this for?

Comment (Hossam-Elgmmal) [reply app/build.gradle.kts:100]: Protobuf is a language-neutral mechanism for serializing structured data. It supports many languages.
Here we choose the required language with
```
register("java") { option("lite") }
register("kotlin") { option("lite") }
```
and add the needed dependency for it with 

```
implementation(libs.protobuf.kotlin.lite)
```
Please, let me know if you'd like me to elaborate further.

Comment (Hossam-Elgmmal) [reply app/build.gradle.kts:100]: I think a better comment for it could be
```
// protobuf - make Kotlin objects based on protobuf schema
```

Comment (qamarelsafadi) [reply app/build.gradle.kts:100]: where do you think we will need it?

Comment (Hossam-Elgmmal) [reply app/build.gradle.kts:100]: The proto datastore needs a `Serializer<T>` and in our case the type `T` will be the generated Java class.
The `Serializer` determines the on-disk format and API for accessing it.
The type `T` handles encoding and decoding the data.

If it is too much for our use cases we can use kotlinx serialization json 
or even drop it all together and use preferences datastore

**Review by aarundn — COMMENTED** [review]
Approved but the latest version of kotlin is 2.0.21

**Comment by Hossam-Elgmmal** [comment]
> Approved but the latest version of kotlin is 2.0.21
we can use Kotlin 2

**Review by qamarelsafadi — APPROVED** [review]

**Review by Mohamed-Ayad902 — APPROVED** [review]

**Pull request merged**